server/plugin.go

```diff
@@ -180,190 +235,428 @@ func (p *Plugin) getGitHubUserInfo(userID string) (*GitHubUserInfo, *APIErrorRes
-		return err
+		p.API.LogWarn("Couldn't get bot's DM channel", "userID", userID, "error", err.Error())
+		return
 	}
 
 	post := &model.Post{
 		UserId:    p.BotUserID,
 		ChannelId: channel.Id,
 		Message:   message,
 		Type:      postType,
-		Props: map[string]interface{}{
-			"from_webhook":      "true",
-			"override_username": GITHUB_USERNAME,
-			"override_icon_url": GITHUB_ICON_URL,
-		},
 	}
 
 	if _, err := p.API.CreatePost(post); err != nil {
-		mlog.Error(err.Error())
-		return err
+		p.API.LogWarn("Failed to create DM post", "userID", userID, "error", err.Error())
+		return
 	}
-
-	return nil
 }
 
-func (p *Plugin) PostToDo(info *GitHubUserInfo) {
-	text, err := p.GetToDo(context.Background(), info.GitHubUsername, p.githubConnect(*info.Token))
+func (p *Plugin) PostToDo(info *BitbucketUserInfo) {
+	text, err := p.GetToDo(context.Background(), info, p.bitbucketConnect(*info.Token))
 	if err != nil {
-		mlog.Error(err.Error())
+		p.API.LogWarn("Failed to get todo text", "userID", info.UserID, "error", err.Error())
 		return
 	}
 
-	p.CreateBotDMPost(info.UserID, text, "custom_git_todo")
+	p.CreateBotDMPost(info.UserID, text, "custom_bitbucket_todo")
 }
 
-func (p *Plugin) GetToDo(ctx context.Context, username string, githubClient *github.Client) (string, error) {
-	config := p.getConfiguration()
+func (p *Plugin) GetToDo(ctx context.Context, userInfo *BitbucketUserInfo, bitbucketClient *bitbucket.APIClient) (string, error) {
+	userRepos, err := p.getUserRepositories(ctx, bitbucketClient)
+	if err != nil {
+		return "", errors.Wrap(err, "error occurred while searching for repositories")
+	}
 
-	issueResults, _, err := githubClient.Search.Issues(ctx, getReviewSearchQuery(username, config.GitHubOrg), &github.SearchOptions{})
+	yourAssignments, err := p.getAssignedIssues(ctx, userInfo, bitbucketClient, userRepos)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "error occurred while searching for assignments")
 	}
 
-	notifications, _, err := githubClient.Activity.ListNotifications(ctx, &github.NotificationListOptions{})
+	yourOpenPrs, err := p.getOpenPRs(ctx, userInfo, bitbucketClient, userRepos)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "error occurred while searching for your open PRs")
 	}
 
-	yourPrs, _, err := githubClient.Search.Issues(ctx, getYourPrsSearchQuery(username, config.GitHubOrg), &github.SearchOptions{})
+	assignedPRs, err := p.getAssignedPRs(ctx, userInfo, bitbucketClient, userRepos)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "error occurred while searching for assigned PRs")
+	}
+
+	text := "##### Your Assignments\n"
+
+	if len(yourAssignments) == 0 {
+		text += "You don't have any assignments.\n"
+	} else {
+		text += fmt.Sprintf("You have %v assignments:\n", len(yourAssignments))
+
+		for _, assign := range yourAssignments {
+			text += getToDoDisplayText(BitbucketBaseURL, assign.Title, assign.Links.Html.Href, "")
+		}
+	}
```

Comment:
I prefer the following in order to drop the else block

```suggestion
	text := "##### Your Assignments\n"

    textAssignments := "You don't have any assignments.\n"
	if len(yourAssignments) != 0 {
		textAssignments = fmt.Sprintf("You have %v assignments:\n", len(yourAssignments))
		for _, assign := range yourAssignments {
			textAssignments += getToDoDisplayText(BitbucketBaseURL, assign.Title, assign.Links.Html.Href, "")
		}
	}
	text += textAssignments
```